admin/services/pin.service.js

```diff
@@ -78,4 +80,14 @@ pinService.expireMultiplePins = async (pupilIds) => {
   return pupilDataService.sqlUpdatePinsBatch(data)
 }
 
+/**
+ * Detects possible check form and check window error that should prevent pin generation
+ * @returns {String}
+ */
+pinService.getPinsOverviewError = async () => {
+  const checkWindow = await checkWindowDataService.sqlFindOneCurrent()
+  const allForms = await checkFormService.getAllFormsForCheckWindow(checkWindow && checkWindow.id)
```

Comment:
Would you really want to call this without an argument?  It calls the data-layer directly - it does not throw.

Reply:
@jon-shipley I just noticed the implementation of it.
Literally only thing it does is call to `sqlFetchSortedActiveFormsByName` data method which seems redundant as a normal service method.
In addition the data layer method returns all the forms that are not deleted if checkWindow id is not provided
 Isn't that wrong?

Reply:
@GeorgeChatzigiannis If you don't have a checkWindow to put in the `getAllFormsForCheckWindow` call we need to throw an error.